ext/promise.js

```diff
@@ -65,10 +65,17 @@ require("../lib/registered-extensions").promise = function (mode, conf) {
 		if (!resolvedMode) resolvedMode = "then";
 
 		if (resolvedMode === "then") {
+			var nextTickFailure = function () {
+				nextTick(onFailure);
+			};
 			promise.then(
 				function (result) { nextTick(onSuccess.bind(this, result)); },
-				function () { nextTick(onFailure); }
+				nextTickFailure
 			);
+			// If `finally` is a function we attach to it to remove cancelled promises.
+			if (typeof promise.finally === "function") {
+				promise.finally(nextTickFailure);
```

Comment:
@Page-  @medikoo won't this always call `onFailure` and always remove it from the cache? It looks like `.finally` is called regardless http://bluebirdjs.com/docs/api/finally.html

Reply:
I see, thanks for the explanation.

The only case I see that could potentially produce a false positive is if one of `nextTick`'s [implementations](https://github.com/medikoo/next-tick/blob/master/index.js) doesn't guarantee in-order execution of the scheduled functions.

Reply:
> The only case I see that could potentially produce a false positive is if one of nextTick's implementations doesn't guarantee in-order execution of the scheduled functions.

I'm pretty sure order is guaranteed in all of them, otherwise there will be dragons :)